Copy edit of 2020 CSS chapter

bazzadp opened the pull request:
Progress on #1432 and #898

Made the following changes:
- Set all graphs to use san-serif on labels for those that don't have Roboto installed
- Added labels to a few graphs
- Changed gradient-with-most-stops to be 1px high reducing it from 14 KB to 1.5 KB — if we're gonna call out the site for using 8 KB of CSS instead of doing this guess we should do it right 😁
- Removed smart quotes (see #1485)
- Lowercase web as per our style guide
- Change "stylesheet" to "style sheet" as per our style guide
- Moved a few commas to em-dashes where sentence already included a lot of commas
- Removed a lot of italics and bold emphasis as felt not needed.
- Added some missing code formatting
- Added a load of TODOs with questions, queries and suggestions for authors. Will call these out below.
- Other standard edits
- Removed unedited tag

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1,87 +1,96 @@
 
-For certain metrics, looking at the CSS AST was not enough. We wanted to look at [SCSS](https://sass-lang.com/) wherever it was provided via sourcemaps as it shows us what developers *need* from CSS that is not yet possible, whereas studying CSS shows us what developers currently use that is. For that, we had to use a *custom metric*, i.e. JS code that runs in the crawler when it visits a given p ...
+For certain metrics, looking at the CSS AST was not enough. We wanted to look at [SCSS](https://sass-lang.com/) wherever it was provided via sourcemaps as it shows us what developers need from CSS that is not yet possible, whereas studying CSS shows us what developers currently use that is. For that, we had to use a *custom metric*—JavaScript code that runs in the crawler when it visits a given p ...
 
-Custom metrics were also used for part of the [Custom properties analysis](#custom-properties). While we can get a lot of information about custom property usage from the stylesheets alone, we cannot build a dependency graph without being able to look at the DOM tree for context, as custom properties are inherited. Looking at the computed style of the DOM nodes also gives us information like what ...
+{# TODO (Authors): it may be obvious to most people who use it (I don't) but is it worth commenting on difference between SCSS and Sass? They are used interchangeably in some places but differentiated in others, so think worth covering here. #}
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

LeaVerou replied:
This is now inaccurate, since we didn't use custom metrics only for SCSS but also for custom properties. The reference to that custom metric appears to now be removed?

bazzadp replied:
Ignore the bit in red. GitHub struggles to show the actual changes when there are large edits like there are here. so concentrate on the bit in green. The custom metrics and custom properties bit is still in there (but in a different place) so fi you want to suggest a change to that then click on the Files tab, find it and click on the line to suggest a change.

In meantime can you comment on the lines in green?

bazzadp replied:
Thanks. I added this, but actually done in the Sass section rather than here.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -94,18 +103,20 @@ Another metric of size is the number of rules. The median page carries a total o
 
 ## Selectors and the cascade
 
+{# TODO (Authors) We try to avoid empty headrs. Could we have a short intro sentence or two here? #}
```
@LeaVerou, @svgeesus, @rachelandrew - could one of you add an intro to this section here?

bazzadp replied:
A went with this. Feel free to shout if you disagree:

> CSS offers a number of ways of apply styles to page, from classes, ids and using the all-important cascade to avoid duplicating styles. So how are developers applying their styling to their pages?

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -115,22 +126,40 @@ Only a few of the top classes were presentational, with most of those being eith
 
 Despite IDs being discouraged these days in some circles due to their much higher specificity, most websites still use them, albeit sparingly. Fewer than half of pages used more than one ID in any of their selectors (had a max specificity of (1,x,y) or less) and nearly all had a median specificity that did not include IDs (0,x,y).
 
+{# TODO Authors—Can you explain this (1,x,y)/(0,x,y) syntax more? #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
MDN explains specificity in terms of "hundreds, tens and ones" so if x was 3 and y was 4 in that example then (1,x,y) would be 1 hundreds, 3 tens, 4 ones - the number 134. But this leads authors to think that, for example, 12 class selectors (x=12) is more specific than one ID selector - 12 *10 = 120 > 100. So CSS experts tend to write the specificity with the number of ID, class, and element/pseudo-element selectors as a list: (ID, class, element).

bazzadp replied:
Is there somewhere we can link to this for other readers who are similarly confused? Don't see any mention of this notation in https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity

LeaVerou replied:
The spec? :) https://www.w3.org/TR/selectors/#specificity-rules

bazzadp replied:
Like it. Added a sentence to this in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1846/commits/d83e68afcf451c0735bddf8b527fb0b578be9435. Marking as resolved but feel free to reopen this and suggest alternative language.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -115,22 +126,40 @@ Only a few of the top classes were presentational, with most of those being eith
 
+{# TODO Authors—Can you explain this (1,x,y)/(0,x,y) syntax more? #}
+
 But what are these IDs used for? It turns out that the most popular IDs are structural: `#content`, `#footer`, `#header`, `#main`, despite [corresponding HTML elements](https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/Document_and_website_structure#HTML_layout_elements_in_more_detail).
 
+{# TODO Authors—This sentence just ends. Do you mean "despite corresponding HTML elements existing that could be used as selectors while also improving the semnatic markup"? #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
Yes, that is a better way to phrase it.

bazzadp replied:
```suggestion
But what are these IDs used for? It turns out that the most popular IDs are structural: `#content`, `#footer`, `#header`, `#main`, despite [corresponding HTML elements](https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/Document_and_website_structure#HTML_layout_elements_in_more_detail) existing that could be used as selectors while also improving the semantic markup.

```

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -115,22 +126,40 @@ Only a few of the top classes were presentational, with most of those being eith
 ) }}
 
 IDs can also be used to intentionally reduce or increase specificity. The specificity hack of writing an ID selector as an attribute selector (`[id="foo"]` instead of `#foo`) was surprisingly rare, with only 0.3% of pages using it at least once. Another ID-related specificity hack, using a negation + descendant selector like `:not(#nonexistent) .foo` instead of `.foo` to increase specificity, was ...
 
-### !important
+{# TODO Authors: Oh clever! I'd never heard of this! Presumably that first one is to reduce specificty? Should we make that clearer? Or link to an article/articles about either or both of these hacks for uninitiated like me? #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
yes - "instead of `#foo`) to reduce specificity was surprisingly rare"

I suggest [specificity hack](https://csswizardry.com/2014/07/hacks-for-dealing-with-specificity/)

bazzadp replied:
Fixed in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1846/commits/c90c8fd4362f57f70b4e331a83bde12d73e4a209

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -229,9 +260,11 @@ When it comes to pseudo-elements, after the usual suspects `::before` and `::aft
 
 ## Values and units
 
+{# TODO authors—Can we have an intro sentence or two so we don't have an empty heading here? #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - could you add something here?

bazzadp replied:
I went with this. Feel free to shout if you disagree:

> CSS provides a number of ways of specifying values and units, either in set lengths or calculations or based on global keywords.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -241,18 +274,20 @@ The humble `px` unit has gotten a lot of negative press over the years. At first
 ) }}
 
-The `px` unit is still going strong as the most popular length unit overall, with a whopping **72.58% of all length values across all stylesheets using `px`**! And if we exclude percentages (since they are not really a unit) the share of `px` increases even more, to 84.14%.
+{# TODO Authors—What do you think about dropping the `<` and `>` signs and code formatting here and using length instead fo `<length>`? I know that's the official syntax of CSS data types (https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Types#Syntax), but IMHO rarely used outside of CSS experts and wonder if the average reader would get confused with HTML elements? I know I did until I looke ...
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
I would be fine with dropping `<>`

LeaVerou replied:
I think given that we link to docs, if anyone is confused they can follow the links. I would vote against dropping the brackets but not a strong objection.

bazzadp replied:
I agree if you kept the code formatting it would be confusing. But I'm talking about dropping both the brackets and code formatting and using the word "length" rather than the type "`<length>`" because, as I say, I don't think this is well known outside of the hardcode CSS spec contributors like yourselves, and for the casual reader it's too easily confused with HTML elements. The fact we only use it on this one type and nowhere else in this document kinda proves that point IMHO.

bazzadp replied:
OK let's close this and leave it as is. Don't think it's a biggie and was just a suggestion.

```suggestion
```

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -405,15 +440,17 @@ Lengths are the only types of CSS values for which we can omit the unit when the
 ### Calculations
 
-When the [`calc()`](https://developer.mozilla.org/en-US/docs/Web/CSS/calc()) function was introduced for performing calculations between different units in CSS, it was a revolution. Previously, only preprocessors were able to accommodate such calculations, but the results were limited to static values and unreliable, since they were missing the dynamic context that is often necessary.
+When the [`calc()`](https://developer.mozilla.org/en-US/docs/Web/CSS/calc()) function was introduced for performing calculations between different units in CSS, it was a revelation. Previously, only preprocessors were able to accommodate such calculations, but the results were limited to static values and unreliable, since they were missing the dynamic context that is often necessary.
+
+{# TODO (authors)—revolution or revelation. I think revelation (aka "a great success") but not 100% sure (could be "A sudden, vast change in a situation, a discipline, or the way of thinking and behaving."): https://wikidiff.com/revolution/revelation #}
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
revolution is correct

LeaVerou replied:
revolution

bazzadp replied:
OK suggesting reverting and removing TODO:

```suggestion
When the [`calc()`](https://developer.mozilla.org/en-US/docs/Web/CSS/calc()) function was introduced for performing calculations between different units in CSS, it was a revolution. Previously, only preprocessors were able to accommodate such calculations, but the results were limited to static values and unreliable, since they were missing the dynamic context that is often necessary.
```

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -451,9 +490,11 @@ It appears that most of this usage is to subtract pixels from percentages, as ev
 Most calculations are very simple, with 99.5% of calculations involving up to 2 different units, 88.5% of calculations involving up to 2 operators and 99.4% of calculations involving one set of parentheses or fewer (3 out of 4 calculations include no parentheses at all).
 
 {# TODO (analysts): Figure out what happened to the 3+ label in this chart. #}
+{# TODO Rick, I couldn't get this to show despite trying a few different things. Should we just remove this cateogry as it's 0 anyway? Might be less confusing. #}
```
@rviscomi any ideas here?

bazzadp replied:
Finally got this to work with a space between the 3 and the +

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -464,35 +505,37 @@ Most calculations are very simple, with 99.5% of calculations involving up to 2
   sql_file="keyword_totals.sql"
 ) }}
 
 The `all` property was [introduced in 2013](https://www.w3.org/TR/2013/WD-css3-cascade-20130103/#all-shorthand) and gained [near-universal support in 2016 (except Edge) and universal support earlier this year](https://caniuse.com/css-all). It is a shorthand of nearly every property in CSS (except custom properties, `direction`, and `unicode-bidi`), and only accepts the four global keywords as val ...
 
+{# TODO authors—what are the four global keywords and should we link them here? The spec only lists `initial`, `inherit` and `unset` but then you talk about `revert` so is that the missing 4th one? Should we explain this discrepency? #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
The original three are `initial`, `inherit` and `unset` [defined in CSS values](https://drafts.csswg.org/css-values-4/#common-keywords), and now also  `revert` [defined in CSS Cascade](https://drafts.csswg.org/css-cascade-4/#valdef-all-revert)

bazzadp replied:
What about this then:

```suggestion
The `all` property was [introduced in 2013](https://www.w3.org/TR/2013/WD-css3-cascade-20130103/#all-shorthand) and gained [near-universal support in 2016 (except Edge) and universal support earlier this year](https://caniuse.com/css-all). It is a shorthand of nearly every property in CSS (except custom properties, `direction`, and `unicode-bidi`), and only accepts the four global keywords (`initial`, `inherit`, `unset`, and `revert`) as values. It was envisioned as a one liner CSS reset, either as `all: unset` or `all: revert`, depending on what kind of reset we wanted. However, adoption is still very low: we only found `all` on 477 pages (0.01% of all pages), and only used with the `revert` keyword.

{# TODO authors—what are the four global keywords and should we link them here? The spec only lists `initial`, `inherit` and `unset` but then you talk about `revert` so is that the missing 4th one? Should we explain this discrepency? #}

```

bazzadp replied:
In case it's not obvious from above with GitHub diff being less than helpful, I just add the bit in parenthesis and removed the TODO - everything else stayed the same:

> ...only accepts the four global keywords (`initial`, `inherit`, `unset`, and `revert`) as values.

bazzadp replied:
Or alternatively just hyperlinking [four global keywords](https://drafts.csswg.org/css-cascade-4/#defaulting-keywords) text with that link?

```suggestion
The `all` property was [introduced in 2013](https://www.w3.org/TR/2013/WD-css3-cascade-20130103/#all-shorthand) and gained [near-universal support in 2016 (except Edge) and universal support earlier this year](https://caniuse.com/css-all). It is a shorthand of nearly every property in CSS (except custom properties, `direction`, and `unicode-bidi`), and only accepts the [four global keywords](https://drafts.csswg.org/css-cascade-4/#defaulting-keywords) as values. It was envisioned as a one liner CSS reset, either as `all: unset` or `all: revert`, depending on what kind of reset we wanted. However, adoption is still very low: we only found `all` on 477 pages (0.01% of all pages), and only used with the `revert` keyword.
```

bazzadp replied:
Or why not both!:

```suggestion
The `all` property was [introduced in 2013](https://www.w3.org/TR/2013/WD-css3-cascade-20130103/#all-shorthand) and gained [near-universal support in 2016 (except Edge) and universal support earlier this year](https://caniuse.com/css-all). It is a shorthand of nearly every property in CSS (except custom properties, `direction`, and `unicode-bidi`), and only accepts the [four global keywords](https://drafts.csswg.org/css-cascade-4/#defaulting-keywords) (`initial`, `inherit`, `unset`, and `revert`) as values. It was envisioned as a one liner CSS reset, either as `all: unset` or `all: revert`, depending on what kind of reset we wanted. However, adoption is still very low: we only found `all` on 477 pages (0.01% of all pages), and only used with the `revert` keyword.
```

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -502,25 +545,28 @@ Similarly, 99.89% of functionally specified sRGB colors are using the since-fore
 So why do people stray from these tried and true formats? To express alpha transparency. This is clear when you look at `rgba()`, which is used 40 times more than `rgb()` (13.82% vs 0.34% of all colors) and `hsla()`, which is used 30 times more than `hsl()` (0.25% vs 0.01% of all colors).
 
-And yes, these numbers also show that despite the much-vaunted (but [largely incorrect](https://drafts.csswg.org/css-color-4/#the-hsl-notation)) *easily-understood* or *easily-modified advantages* of HSL, in practice it is used *far less* than RGB.
+And yes, these numbers also show that despite the much-vaunted (but [largely incorrect](https://drafts.csswg.org/css-color-4/#the-hsl-notation)) easily-understood or easily-modified advantages of HSL, in practice it is used far less than RGB.
+
+{# TODO Authors—finding it quite difficult to understand that sentence. So it's much-vaulted as easily understood but it's not actually easily understood? #}
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
OK here is another run at that.

"HSL is widely supposed to be easy to understand and easy to modify. But these numbers show that in practice, HSL is used in style sheets far less than RGB. Also, those advantages are [greatly over-stated](https://drafts.csswg.org/css-color-4/#ex-hsl-sucks)"

bazzadp replied:
LGTM to me. You can sue the little document with the +/- symbol to make suggestions which can then be accepted here (including removing the TODO):

```suggestion
HSL is widely supposed to be easy to understand and easy to modify. But these numbers show that in practice, HSL is used in style sheets far less than RGB. Also, those advantages are [greatly over-stated](https://drafts.csswg.org/css-color-4/#ex-hsl-sucks).
```

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -669,15 +715,19 @@ And if you are going for fanciful color names, why not define your own with CSS
+And, lastly, the once-deprecated—now partially un-deprecated—system colors like `Canvas` and `ThreeDDarkShadow`: these were a terrible idea, introduced to emulate the typical user interface of things like Java or Windows 95, and already unable to keep up with Windows 98, they soon fell by the wayside. Some sites use these system colors to try and fingerprint you, a loophole that [we are trying to ...
 
 The rather useful value `currentColor`, surprisingly, trailed at 0.14% of all sRGB colors (1.62% of all named colors).
 
-All the colors we discussed so far have one thing in common: sRGB, the standard color space for the Web (and for High Definition TV, which is where it came from). Why is that so bad? Because it can only display a limited range of colors: your phone, your TV, and probably your laptop are able to display much more vivid colors due to advances in display technology. Displays with wide color gamut, w ...
+{# TODO authors—Is it worth explaining why it's useful or is that obvious? #}
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
I honestly thought it was obvious (but I have the curse of knowledge here). There are lots of colors that we can see, but you can't use them in web pages. Obviously we want to, especially since native apps can use them.

bazzadp replied:
I think this is another case where GitHub trying it's best to show the replacement text in red is causing more confusion than help! Ignore the bit in red - I'm asking why `currentColor` is useful and whether that should be explained? If it's the current color why do you need this? Maybe a link to this https://css-tricks.com/currentcolor/ would cover it?

svgeesus replied:
Aha! Yes, that link is a good description of why it s useful, compared to just duplicating a color in multiple places.

bazzadp replied:
Added that link in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1846/commits/57d05764a596373118a1e666231f9c1e0b0bfca8

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -669,15 +715,19 @@ And if you are going for fanciful color names, why not define your own with CSS
+And we are still missing out. Despite being [implemented in Safari in 2016](https://webkit.org/blog/6682/improving-color-on-the-web/), the use of display-p3 color in web pages is vanishingly small. Our crawl of the web found only 29 mobile and 36 desktop pages using it! (And more than half of those were syntax errors, mistakes, or attempts to use the never-implemented `color-mod()` function). We  ...
 
-Compatibility, right? You don’t want things to break? No. In the stylesheets we examined, we found solid use of fallback: with document order, the cascade, `@supports`, the `color-gamut` media query, all that good stuff. So in a style sheet we would see the color the designer wanted, expressed in display-p3, and also a fallback sRGB color. We computed the visible difference (a calculation called  ...
+Compatibility, right? You don't want things to break? No. In the style sheets we examined, we found solid use of fallback: with document order, the cascade, `@supports`, the `color-gamut` media query, all that good stuff. So in a style sheet we would see the color the designer wanted, expressed in display-p3, and also a fallback sRGB color. We computed the visible difference (a calculation called ...
+
+{# TODO Authors—you're completely losing me with this :-) Not sure if there's a way to simplify it more or if this is just the nature of this topic and it's just more advanced, which is fine — the rest of the chapter so far is is more than readable for those less-advance amongst us! But still not sure what your point is here? They shouldn't have used display-p3 because sRGB was good enough? Or is ...
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
Yes, it is a bit dense, but unpacking it more and giving more tutorial material would make it much longer.

> Or is it more than people are just cautiously experimenting at the moment so no real gains yet, but more to come

Yes, that is it.

I am actually considering making a whole blog post about just this one point. But that will take me some time.

bazzadp replied:
What do you think about adding that sentence to the end then?

```suggestion
Compatibility, right? You don't want things to break? No. In the style sheets we examined, we found solid use of fallback: with document order, the cascade, `@supports`, the `color-gamut` media query, all that good stuff. So in a style sheet we would see the color the designer wanted, expressed in display-p3, and also a fallback sRGB color. We computed the visible difference (a calculation called [ΔE2000](https://zschuessler.github.io/DeltaE/learn/)) between the desired and fallback color and this was typically quite modest. A small tweak. A careful exploration. In fact, 37.6% of the time, the color specified in display-p3 actually fell inside the range of colors (the gamut) that sRGB can manage. It seems people are just cautiously experimenting with thus at the moment rather than to get real gains, but more is surely to come so one to watch.
```

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -929,36 +977,39 @@ Prefixed gradients are also still very common, even though prefixes haven't been
+  height=122
 ) }}
 
 ## Layout
 
+{# TODO Authors:again can we get an intro sentence or two? #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - could you add something here?

bazzadp replied:
I went with this. Feel free to shout if you disagree:

> CSS now has a number of layout options—a far cry from the tables when tables had to be used for layouts. Flexbox, Grid and Multiple-column layouts are now well supported in most browsers so let's look at how these are being used.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1059,30 +1113,32 @@ We were glad to discover that most of these transitions are fairly short, with t
   sql_file="transition_animation_names.sql"
 ) }}
 
 ## Visual effects
 
+{# TODO authors — intro sentence #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - Could you add something here?

bazzadp replied:
I went with this. Feel free to shout if you disagree:

> CSS also offers a huge variety of visual effects giving designers access to advanced design techniques built into browsers that can be accessed with small amounts of code.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1241,21 +1296,36 @@ Examining the selectors on which custom properties are declared further confirms
 
 ## CSS and JS
 
+{# TODO Authors:intro sentence #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - could you add something here?

bazzadp replied:
I went with this. Feel free to shout if you disagree:

> The last few years has seen a greater interaction between CSS and JavaScript, beyond the simple setting of CSS classes and styles or off. So how much are we using technologies like Houdini and techniques like CSS-in-JS?

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1241,21 +1296,36 @@ Examining the selectors on which custom properties are declared further confirms
 ### Houdini
 
-You have likely heard of Houdini by now. Since [several Houdini specs have shipped in browsers](https://ishoudinireadyyet.com/), we figured it's time to see if they are actually used in the wild yet. Short answer: no. And now for the longer answer…
+You have likely heard of Houdini by now. Since [several Houdini specs have shipped in browsers](https://ishoudinireadyyet.com/), we figured it is time to see if they are actually used in the wild yet. Short answer: no. And now for the longer answer…
+
+{# TODO Authors: for those that haven't heard of it should we give a brief intro? Maybe in above empty header? #}
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

bazzadp replied:
Thanks, added.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1268,27 +1338,42 @@ There is so much discussion (or argument) about CSS-in-JS that one could assume
 
 ## Internationalization
 
+{# TODO Authors: Intro sentence #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - could you add something here?

svgeesus replied:
Hmm. OK here is an intro sentence.

"English, like many languages, is written in horizontal lines and the characters are laid out from left to right. But some languages (such as Arabic and Hebrew) are mostly written right to left. And then there are languages which may be written in vertical lines, from top to bottom. Not to mention quotations from other languages. So things can get quite complicated. Both HTML and CSS have ways to handle this."

bazzadp replied:
Added that in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1846/commits/d83e68afcf451c0735bddf8b527fb0b578be9435. Marking as resolved.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1268,27 +1338,42 @@ There is so much discussion (or argument) about CSS-in-JS that one could assume
 
 ## Internationalization
 
+{# TODO Authors: Intro sentence #}
+
 ### Direction
 
-When text is presented in horizontal lines, most writing systems display characters from left to right. Urdu, Arabic and Hebrew display characters from right to left, except for  numbers, which are written from left to right; they are bidirectional. Some characters  — such as brackets, quote marks, punctuation — could be used in either a left to right or a right to left context, and are said to b ...
+When text is presented in horizontal lines, most writing systems display characters from left to right. Urdu, Arabic and Hebrew display characters from right to left, except for numbers, which are written from left to right; they are bidirectional. Some characters—such as brackets, quote marks, punctuation—could be used in either a left to right or a right to left context and are said to be direc ...
+
+To support bidirectionality, explicit support for indicating direction is available in both HTML via ([the `dir` attribute](https://html.spec.whatwg.org/multipage/dom.html#the-dir-attribute) and the [`<bdo>` element](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-bdo-element)), and CSS (the [direction](https://www.w3.org/TR/css-writing-modes-3/#direction) and [`unicode-bidi` ...
 
-To support bidirectionality,  explicit support for indicating direction is available in both HTML via ([the `dir` attribute](https://html.spec.whatwg.org/multipage/dom.html#the-dir-attribute) and the [`<bdo>` element](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-bdo-element)), and CSS (the [direction](https://www.w3.org/TR/css-writing-modes-3/#direction) and [`unicode-bidi ...
-Only 12.14% of pages on mobile (and a similar 10.76% on desktop) set the `dir` attribute on the `<html>` element. Which is fine: most writing systems in the world are **ltr**, and the default `dir` value is **ltr**. Of the pages which *did* set `dir` on `<html>`, 91% set it to `ltr` while 8.5% set it to `rtl` and 0.32% to `auto` (the explicit direction is unknown value, mainly useful for template ...
+Only 12.14% of pages on mobile (and a similar 10.76% on desktop) set the `dir` attribute on the `<html>` element. Which is fine: most writing systems in the world are `ltr`, and the default `dir` value is `ltr`. Of the pages which did set `dir` on `<html>`, 91% set it to `ltr` while 8.5% set it to `rtl` and 0.32% to `auto` (the explicit direction is unknown value, mainly useful for templates whic ...
 
-Why set direction using HTML attributes rather than CSS styling? One reason is separation of concerns: direction has to do with content which is the purview of HTML. It is also the [recommended practice](https://www.w3.org/International/tutorials/bidi-xhtml/index.en): *“Avoid using CSS or Unicode control codes for managing direction where you can use markup”*. After all, the stylesheet might not  ...
+{# TODO Authors: I can't find this 2.63% number in the data (https://docs.google.com/spreadsheets/d/1sMWXWjMujqfAREYxNbG_t1fOJKYCA6ASLwtz4pBQVTw/edit#gid=368716708) and it's not smaller than 0.32% so bit confused??? #}
```
@LeaVerou, @svgeesus, @rachelandrew - thoughts on this?

svgeesus replied:
Um, I don't immediately find it either. Will check more when I have time, I need to start packing for  travel soon, sorry!

bazzadp replied:
@svgeesus did you get a chance to look at this?

bazzadp replied:
I found it @svgeesus . Had to pivot the table to add up all the values and then devide by the total number of pages crawled. Add the table [in the sheet](https://docs.google.com/spreadsheets/d/1sMWXWjMujqfAREYxNbG_t1fOJKYCA6ASLwtz4pBQVTw/edit#gid=368716708) in case anyone else goes looking at changed:

> An even smaller number, 2.63%, set `dir` on `<body>`

to:

> An even smaller number, 2.63%, set `dir` on `<body>` than on `<html>`

to make it more obvious were talking about the whole HTML numbers and not just the 0.32 that use auto.

Resolving.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1268,27 +1338,42 @@ There is so much discussion (or argument) about CSS-in-JS that one could assume
-While these properties are fairly [well supported](https://caniuse.com/css-logical-props) (with some exceptions), they are not used very much outside of user agent stylesheets. None of the logical properties were used on more than 0.6% of pages. Most usage was to specify margins and paddings. Logical keywords for `text-align` were used on 2.25% of pages, but apart from that, none of the other key ...
+While these properties are fairly [well supported](https://caniuse.com/css-logical-props) (with some exceptions), they are not used very much outside of user agent style sheets. None of the logical properties were used on more than 0.6% of pages. Most usage was to specify margins and paddings. Logical keywords for `text-align` were used on 2.25% of pages, but apart from that, none of the other ke ...
 
 ## Browser support
 
+{# TODO Authors: another empty heading #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - could you add something here?

bazzadp replied:
I went with this. Feel free to shout if you disagree:

> A perennial problem with the web platform is how to introduce new features and extend the platform. CSS has seen us moving from vendor prefixes to feature queries as a better way of introducing change so we wanted to look at how those two techniques were being used.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1398,29 +1477,31 @@ Overall, `-*-min-pixel-ratio` comprises three quarters of prefixed media feature
   sql_file="supports_criteria.sql"
 ) }}
 
 ## Meta
 
+{# TODO Authors: intro paragraph please #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - could you add something here?

bazzadp replied:
I went with this. Feel free to shout if you disagree:

> Up until now we've looked at what CSS developers have used, but in this section we want to look more about _how_ they are using it.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1581,16 +1659,18 @@ Did you know that `grid-template-columns`, `grid-template-rows`, and `grid-templ
 
 ### CSS mistakes
 
+{# TODO Authors: Intro sentence or two #}
+
```
@LeaVerou, @svgeesus, @rachelandrew - could you add something here?

bazzadp replied:
I went with this. Feel free to shout if you disagree:

> As with any complex, evolving platform not everything is done correctly. So let's look at some of the mistakes developers are making out there.

bazzadp reviewed (commented):
Highlighted the TODOs for your comment.

BTW forgot to say - fantastic chapter. Very interesting and very readable even for someone like me with limited CSS knowledge (other than the colour gamut stuff that went completely over my head 😀)! So really well done given the amount you covered!

Let me know your thoughts on the points I've highlighted with TODOs and feel free to suggest changes via GitHub or to just push suggestions directly to this branch.

svgeesus commented:
I notice that the advance release of the almanac used the [small color table](https://github.com/LeaVerou/css-almanac/blob/master/media/pairs-dE.html) rather than the [much bigger table](https://github.com/LeaVerou/css-almanac/blob/master/media/pairs-dE-large.html) from the >100k re-run. The difference is significant, the first run found only 23 unique pairs of (P3, sRGB) colors while the re-run found 114.
The graphic used is the correct ne, with all 114 color pairs plotted.

svgeesus commented:
> (other than the colour gamut stuff that went completely over my head 😀)!

Sorry about that. So, this was the explanation I used (trying to keep it short, but apparently it wasn't understandable enough):

> All the colors we discussed so far have one thing in common: sRGB, the standard color space for the Web (and for High Definition TV, which is where it came from). Why is that so bad? Because it can only display a limited range of colors: your phone, your TV, and probably your laptop are able to display much more vivid colors due to advances in display technology. Displays with wide color gamut, which used to be reserved for well-paid professional photographers and graphic designers, are now available to everyone. Native apps use this capability, as do digital movies and streaming TV services, but until recently the Web was missing out.

Gamut is "the range of colors that can be displayed" and a narrower gamut means less colors (omitting the more vivid ones). Suggestions to re-word that would be welcome.

I felt it was important to capture this data, despite p3 colors being barely used, because this is expected to increase over then next few years and therefore this year is the baseline to compare against.

bazzadp commented:
> I notice that the advance release of the almanac used the [small color table](https://github.com/LeaVerou/css-almanac/blob/master/media/pairs-dE.html) rather than the [much bigger table](https://github.com/LeaVerou/css-almanac/blob/master/media/pairs-dE-large.html) from the >100k re-run. The difference is significant, the first run found only 23 unique pairs of (P3, sRGB) colors while the re-run found 114.
> The graphic used is the correct ne, with all 114 color pairs plotted.

Could we link to this instead? I feel the full table would be quite large, and having it in a spreadsheet makes it easier to copy and paste and use fiddle with the data if you want.

Plus there's a bug in our PDF generator that doesn't like tables spanning more than one page 😀

I've copied the data to your spreadsheet and added a link to figcaption in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1846/commits/97d3ed5360563056dd9e1883fd82d9e9ef809995 and you can see what that looks like here: https://20201228t110002-dot-webalmanac.uk.r.appspot.com/en/2020/css#fig-29

WDYT?

> Or was it the chromaticity diagram or the use of deltaE that was unclear?
> 
> I felt it was important to capture this data, despite p3 colors being barely used, because this is expected to increase over then next few years and therefore this year is the baseline to compare against.

I get the basic concept but the finer details were a little lost on me, but that's OK. What's more concerning is I didn't really get the point you were trying to make. Did you see my comment in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1846#discussion_r549166250? 

You may have to click this to see the rest of my comments:
![See more button](https://user-images.githubusercontent.com/10931297/103210383-63030480-48fd-11eb-846f-2983894093f8.png)

Here's what I said:

> {# TODO Authors—you're completely losing me with this :-) Not sure if there's a way to simplify it more or if this is just the nature of this topic and it's just more advanced, which is fine — the rest of the chapter so far is is more than readable for those less-advance amongst us! But still not sure what your point is here? They shouldn't have used display-p3 because sRGB was good enough? Or is it more than people are just cautiously experimenting at the moment so no real gains yet, but more to come (as covered by your next bit of writing after this table? #}

svgeesus commented:
> What's more concerning is I didn't really get the point you were trying to make. Did you see my comment in #1846 (comment)?

I did now :)

svgeesus commented:
What is the timescale for reviewing these comments?

bazzadp commented:
> What is the timescale for reviewing these comments?

Depends how keen you are to remove the "Unedited" label 😁 sooner would be better rather than later though as now is when it gets a lot of views.

We can have a look after the majority are answered/resolved/not doing and decide if it's basically edited now and and leave one or two till later.

LeaVerou commented:
> Removed a lot of italics and bold emphasis as felt not needed.

Italics and bold help chunking, which makes text easier to process than one blob of uniform text and more pleasant to read. They are the typographic equivalent of formatting a long phone number with dashes and parentheses instead of presenting it as a series of numbers. It is not a good idea to remove them.

bazzadp commented:
> > Removed a lot of italics and bold emphasis as felt not needed.
> 
> Italics and bold help _chunking_, which makes text easier to process than one blob of uniform text, and more pleasant to read. They are the typographic equivalent of formatting a long phone number with dashes and parentheses instead of presenting it as a series of numbers. It is not a good idea to remove them.

Personally I disagree - that's what punctuation is for. Italics and bold are for emphasis and (IMHO) they are often overused for this. Some [style guides agree](https://wmich.edu/writing/about), though the Google Developer Documentation Style Guide [isn't quite a clear on the subject as that](https://developers.google.com/style/text-formatting?hl=en).

For the Web Almanac we use italics to introduce a new technical term and (very sparingly) for emphasis. We also have the "big figures" for emphasising important or interesting stats and this is better to me in that it also links back to the data and queries to back that up.

Now there is a balance between celebrating the fact we have different, and diverse authors (and so writing style!) for each chapter, and trying to have a consistency across all the chapters. That's something we try to address in the editing process and yes it does lead to tough calls and disagreement when writing styles clash. I feel this is one point we should be consistent on to avoid confusing readers when one chapter uses bold and italics for one reason, and another for another. If you look across the other chapters bold and italics are not used much at all (and only sometimes has that been because editors have removed them!)

But more than willing to admit I could have got it wrong in places, so if there's particular ones you want to bring back then please add comment in Files tab and we can discuss, but I'd personally be against bringing back all the ones I removed as I really do feel they distracted, more than helped.

I'd be interested to hear others opinion on this too (as long as you three authors don't gang up on this one poor, wannabe editor! 😀) — especially @rachelandrew given her professional editing experience.

bazzadp commented:
@svgeesus @LeaVerou @rachelandrew keen to close this out and removed the Unedited tag from the label. Could you look over the remaining open issues and see if we can merge this soon?

rachelandrew commented:
@bazzadp I'll take a look at any of these things that relate to my bits.

With regard to extensive use of bold and italics, I'm not a fan, and remove them when editing. In general, if a paragraph is so long that you need to add bold to "chunk" it, it might be better to split it into two sections, add a heading or example. Quite a lot of people also find italicised text difficult to read in large blocks. 

Bold and italic also infer some kind of meaning, but what? If the section needs calling out in some way it would be better to have an option to highlight it (something to make it a note, or warning perhaps). When I'm editing, if I see a section in bold, that's a flag for me to ask the author why they thought that bit was so important it should be in bold. Perhaps instead they could explain why it was so important, rather than just wave an important flag by way of emboldening the text.

bazzadp commented:
Have you had a chance to look at this yet @rachelandrew ? I really would like to close this out ASAP even if it means merging this PR and addressing the last few items in future.

rachelandrew commented:
@bazzadp I didn't see anything related to the sections I had worked on. Have I missed something?

bazzadp commented:
Sorry I'm not aware of who worked on what, so was only following up on your comment when you said you'd close out anything related to your bits.

There's still a few unresolved items above but guess they are for @LeaVerou or @svgeesus then.

From your end I presume you're happy to merge this @rachelandrew and that I haven't butchered your writing too much? 😀

LeaVerou commented:
> @bazzadp I'll take a look at any of these things that relate to my bits.
> 
> With regard to extensive use of bold and italics, I'm not a fan, and remove them when editing. In general, if a paragraph is so long that you need to add bold to "chunk" it, it might be better to split it into two sections, add a heading or example. Quite a lot of people also find italicised text difficult to read in large blocks.
> 
> Bold and italic also infer some kind of meaning, but what? If the section needs calling out in some way it would be better to have an option to highlight it (something to make it a note, or warning perhaps). When I'm editing, if I see a section in bold, that's a flag for me to ask the author why they thought that bit was so important it should be in bold. Perhaps instead they could explain why it was so important, rather than just wave an important flag by way of emboldening the text.

I think we may be talking about different things. I do not support overuse of bold and italics, as they defeat the purpose and dilute their meaning. However, when used appropriately, they can increase scannability, reduce cognitive load, and make a paragraph easier to process. I also did not advocate for entire sections in bold.

bazzadp commented:
Thanks @LeaVerou appreciate it. It would be good to get this closed out so I can stop pestering you all! Really do appreciate the huge effort that’s gone into this already but this should hopefully be the last piece!

As I said earlier I’m happy to reintroduce any of the bolds and italics you feel are necessary if you want to make those edits in this branch. Maybe I was too aggressive removing some of them but honesty didn’t think at least some of them were necessary and think Rachel’s covered some of my thinking here. But I’ve been wrong before (and definitely will be again!) and it’s your names at the top, so want you to be comfortable with any edits made and revert any you’re not comfortable with.

bazzadp commented:
Oh and huge congrats on your election to W3C Tag!! Very well deserved!

LeaVerou commented:
I just went through and left comments.

I'm not sure what kind of intro text to add to prevent adjacent headings. I suppose I'm not used to adding blurbs with no purpose just for cosmetic reasons. However, I'm fine with whatever you or anyone else wants to add.

Another issue with removing italic and bold is that it makes the implied voice of the text flat and boring. 
There is a reason why emphasis (`<em>`) in HTML is styled as italic by default: italic denotes *stress*. Emphasizing (appropriate) parts of your sentences while speaking makes for a more engaging talk, while keeping a constant tone of voice makes for a dull one. It's the same with writing.

Compare this: 

> While analyzing CSS code tells us what CSS developers *are* doing, looking at preprocessor code can tell us a bit about what CSS developers *want* to be doing, but *can't*.

with this:

> While analyzing CSS code tells us what CSS developers are doing, looking at preprocessor code can tell us a bit about what CSS developers want to be doing, but can't.

Read it out. Do you see the difference in tone?

I also disagree with a few of the style guide decisions, but didn't leave comments on those as I suspect it's a losing battle. But for the record:
- A "web" is something spiders make. *The* Web which is short for "World Wide Web" is a unique entity, a proper noun, and needs to be capitalized to distinguish it from the common noun [see also: [MLA](https://style.mla.org/the-capitalization-of-web/#:~:text=%E2%80%9CThe%20World%20Wide%20Web%E2%80%9D%20is,short%20form%20of%20the%20name.)]
- "stylesheet" is far more common usage than "style sheet". 17 million results on Google for "stylesheet", 7 million for "style sheet". I've never seen anyone calling them "style sheets" except when explaining what CSS stands for. Even in HTML, it's `rel="stylesheet"`, not `rel="style sheet"`.

LeaVerou commented:
> Oh and huge congrats on your election to W3C Tag!! Very well deserved!

Thank you!

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -1185,16 +1226,16 @@ While at first glance this is impressive adoption, it appears that a major drive
 ) }}
 
-Out of the 1,000 top property names, fewer than 13 are ‘custom’, as in made up by individual web developers.The vast majority are associated with popular software, such as WordPress, Elementor, and Avada. To determine this, we took into account not only which custom properties appear in what software (by searching on GitHub), but also which properties appear in groups with similar frequencies. Th ...
+Out of the 1,000 top property names, fewer than 13 are 'custom', as in made up by individual web developers. The vast majority are associated with popular software, such as WordPress, Elementor, and Avada. To determine this, we took into account not only which custom properties appear in what software (by searching on GitHub), but also which properties appear in groups with similar frequencies. T ...
```
This was a result of converting the smart quotes as discussed in #1485. They were single quotes before my edit but typographical single quotes. what do you suggest here - double quotes?

LeaVerou replied:
I know, but I don't see a reason they should be single quotes. Do you?

bazzadp replied:
Sorry I'm a bit confused what you are asking. I didn't write them as single quotes so presume you did? All I did was convert them from single typographical quotes to single straight quotes.

So are you suggesting we remove the quotes altogether? Or change them to double quotes? Either works for me.

bazzadp replied:
Changed it to double quotes as see this as an air quotes thing.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -94,43 +103,57 @@ Another metric of size is the number of rules. The median page carries a total o
 ### IDs
 
-Despite IDs being discouraged these days in some circles due to their much higher specificity, most websites still use them, albeit sparingly. Fewer than half of pages used more than one ID in any of their selectors (had a max specificity of (1,x,y) or less) and nearly all had a median specificity that did not include IDs (0,x,y).
+Despite IDs being discouraged these days in some circles due to their much higher specificity, most websites still use them, albeit sparingly. Fewer than half of pages used more than one ID in any of their selectors (had a max specificity of (1,x,y) or less) and nearly all had a median specificity that did not include IDs (0,x,y). See the [selector's specification](https://www.w3.org/TR/selectors ...
```
Fixed!

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -669,15 +701,17 @@ And if you are going for fanciful color names, why not define your own with CSS
+The [rather useful value `currentColor`](https://css-tricks.com/currentcolor/), surprisingly, trailed at 0.14% of all sRGB colors (1.62% of all named colors).
 
-All the colors we discussed so far have one thing in common: sRGB, the standard color space for the Web (and for High Definition TV, which is where it came from). Why is that so bad? Because it can only display a limited range of colors: your phone, your TV, and probably your laptop are able to display much more vivid colors due to advances in display technology. Displays with wide color gamut, w ...
+{# TODO authors—Is it worth explaining why it's useful or is that obvious? #}
```
Yeah that link was added after discussion above, but forgot to remove the TODO. Have removed the TODO now.

bazzadp commented on src/content/en/2020/css.md:
```diff
@@ -502,25 +533,26 @@ Similarly, 99.89% of functionally specified sRGB colors are using the since-fore
 So why do people stray from these tried and true formats? To express alpha transparency. This is clear when you look at `rgba()`, which is used 40 times more than `rgb()` (13.82% vs 0.34% of all colors) and `hsla()`, which is used 30 times more than `hsl()` (0.25% vs 0.01% of all colors).
 
-And yes, these numbers also show that despite the much-vaunted (but [largely incorrect](https://drafts.csswg.org/css-color-4/#the-hsl-notation)) *easily-understood* or *easily-modified advantages* of HSL, in practice it is used *far less* than RGB.
+HSL is supposed to be [easy to understand and easy to modify](https://drafts.csswg.org/css-color-4/#the-hsl-notation). But these numbers show that in practice, HSL is used in style sheets far less than RGB, likely because those advantages are [greatly over-stated](https://drafts.csswg.org/css-color-4/#ex-hsl-sucks).
```
Yes it may be a little drier I admit, but as per [the discussion above](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1846#discussion_r549166171) I found the initial text quite difficult to understand:

So it's much-vaulted as easily understood but it's not actually easily understood?

bazzadp commented:
> I'm not sure what kind of intro text to add to prevent adjacent headings. I suppose I'm not used to adding blurbs with no purpose just for cosmetic reasons. However, I'm fine with whatever you or anyone else wants to add.

I made an attempt in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1846/commits/34ba57b90a205f8c6dcd9862d222e02afa84d9af

Empty or "Stacked headings" are seen as bad in writing for a number of reasons. If a number of topics are related under a heading, then there should be some relationship that can explain them. If they are unrelated then we should question if they belong under the same heading.

We can also inform the reader what is in this section without them having to refer to the index/table of contents so they know what will be covered in this section.

And finally, if you click on a heading, and immediately go to another heading then why not just go straight there?

So I tried to add some headings as I saw fit and hopefully you're OK with them. The one I did struggle with was "Meta" as wasn't use what this section was about (hence why an intro paragraph explaining it would be good! 😀) but went with:

> Up until now we've looked at what CSS developers have used, but in this section we want to look more about _how_ they are using it.

With added emphasis on _how_ just for you! 😁

> Another issue with removing italic and bold is that it makes the implied voice of the text flat and boring.
> There is a reason why emphasis (`<em>`) in HTML is styled as italic by default: italic denotes _stress_. Emphasizing (appropriate) parts of your sentences while speaking makes for a more engaging talk, while keeping a constant tone of voice makes for a dull one. It's the same with writing.
> 
> Compare this:
> 
> > While analyzing CSS code tells us what CSS developers _are_ doing, looking at preprocessor code can tell us a bit about what CSS developers _want_ to be doing, but _can't_.
> 
> with this:
> 
> > While analyzing CSS code tells us what CSS developers are doing, looking at preprocessor code can tell us a bit about what CSS developers want to be doing, but can't.
> 
> Read it out. Do you see the difference in tone?

I do to a point, however I think the sentence gets that point across without the need of this emphasis particularly on all three words - the meaning is totalling apparent with the sentence. If you read the sentence out loud, and emphasis all three words you'll be quote out of breath by the end of the sentence.

We use italics to emphasis a new term, and very sparingly for emphasis. I left a few of those in, and added a few back in the last edit after re-reviewing, but honestly don't think it's needed in the above sentence.

And bold is a very strong emphasis - like shouting it. So it should be used very, very sparingly IMHO.

> I also disagree with a few of the style guide decisions, but didn't leave comments on those as I suspect it's a losing battle. But for the record:
> 
> * A "web" is something spiders make. _The_ Web which is short for "World Wide Web" is a unique entity, a proper noun, and needs to be capitalized to distinguish it from the common noun [see also: [MLA](https://style.mla.org/the-capitalization-of-web/#:~:text=%E2%80%9CThe%20World%20Wide%20Web%E2%80%9D%20is,short%20form%20of%20the%20name.)]

I used to be with you on this and when I wrote my book I capitalised this for the reason you give but had so succumb to the publishers style guide on this. However I've changed my mind on this recently. The web is now a common, indicative part of our lives it no longer needs proper noun status in my mind and many style guides are changing to this ([including the AP style guide](https://www.wired.com/2016/04/ap-finally-realizes-2016-will-let-us-stop-capitalizing-internet/). But there's definitely still some debate about this and it does upset some people. At the end of the day, both are used, and I think consistency is more important, and lowercase is what we've used in other chapters.

> * "stylesheet" is far more common usage than "style sheet". 17 million results on Google for "stylesheet", 7 million for "style sheet". I've never seen anyone calling them "style sheets" except when explaining what CSS stands for. Even in HTML, it's `rel="stylesheet"`, not `rel="style sheet"`.

This one really surprised me and I actually looked it up as part of this copy edit and added it to our style guide. The [Google Style Guide we use, says it uses this standard because that's what the W3C say](https://developers.google.com/style/word-list#style-sheet)! As some of you are in with them maybe we could get some definitive answers from them! I guess it is CSS and not CS which is one argument 😀

But it does seem to me that even then it's not clear cut: Doing a Google for `"style sheet" site:https://www.w3.org/` returns 8,060 results versus 7,950 for `"stylesheet" site:https://www.w3.org/` - and I suspect the majority of the ones with spaces are for `Cascading Style Sheet`.

So let's change here and use "stylesheet" without a space. It's what I would have used personally until I looked this up, and you backing this too convinced me. We didn't really have this standardised before (as I say I only looked it up as part of this copyedit) and doing a search the spaceless version is far more common in our codebase. Will update our style guide.

bazzadp commented:
I think this is good to merge now, so will do that before next release unless I hear back from you on anything.
